### DISCUSSION TITLE
ContractFunctionExecutionError: The contract function "name" returned no data ("0x").

### DISCUSSION THREAD

**Discussion by minovision** [wevm/wagmi#2821] [Q&A]
I am able to read onchain data from contracts deployed several days/weeks ago, however when I attempt to read from a newly deployed contract i get the error. Ive experienced this outcome on both mumbai and sepolia.

I redeployed an existing contract and verified i can read from the older one (deployed early last week) but not the newer one

Note that i can read if the call is native ethers, but not via wagmi useContractRead

Is anyone experiencing these issues?

**Comment by minovision** [comment]
the above is being experienced with the following pkgs

    "connectkit": "^1.5.0",
    "viem": "^1.4.2",
    "wagmi": "^1.3.9"

**Comment by ummartoor** [comment]
+1

**Comment by owieth** [comment]
I've also encountered this problem but with an old contract (deployed ~ 10 days ago), locally it fails, on production not but if I try to push a new change, then the error will also happen during the build (vercel).

I'm using the following deps: 

```
"@rainbow-me/rainbowkit": "^1.0.10",
"wagmi/core": "^1.3.10",
"viem": "^1.9.3",
"wagmi": "^1.3.11"
```

**Comment by nsnull0** [comment]
me too also similiar issue with latest wagmi 
```
    "@web3modal/wagmi": "^3.5.0",
    "viem": "^1.20.2",
    "wagmi": "^1.4.12"
```

**Comment by nsnull0** [comment]
I guess can't use the method for local testing chain like ganache since this method somehow only succeed if the contract is verified first.
only verified contract on live production chain will success 😢
